contrib/win32/win32compat/w32-sshfileperm.c

```diff
@@ -178,37 +182,39 @@ check_secure_file_permission(const char *input_path, struct passwd * pw, int rea
 * Check the owner of the file is one of these types: Local Administrators groups or system account
 * Check the users have access permission to the file don't violate the following rules:
 	1. no user other than local administrators group and system account have write permission on the folder
-* Returns 0 on success and -1 on failure
+* Logs a message if the rules are violated, but does not prevent further execution
 */
-int
+void
 check_secure_folder_permission(const wchar_t* path_utf16, int read_ok)
 {
 	PSECURITY_DESCRIPTOR pSD = NULL;
 	PSID owner_sid = NULL, ti_sid = NULL;
 	PACL dacl = NULL;
 	DWORD error_code = ERROR_SUCCESS;
-	BOOL is_valid_sid = FALSE, is_valid_acl = FALSE;
+	BOOL is_valid_sid = FALSE, is_valid_acl = FALSE, is_first = TRUE;
 	wchar_t* bad_user = NULL;
 	int ret = 0;
+	size_t log_msg_len = (DNLEN + 1 + UNLEN) * 2 + 3; // +3 for ", " and null terminator
```

Comment:
My preference is to use `const` for the extra lengths instead of a comment as code and comments can get out of sync (someone modifies the code, but doesn't update the comment).  So:

```c
const size_t NULL_TERMINATOR_LEN = 1;
const size_t COMMA_SPACE_LEN = 2;
```

This applies elsewhere as well.